### PR TITLE
Dedup slurm infra + Add slurm support to mlagility

### LINKED ISSUE #89: Implement --use-slurm
* Implement `--use-slurm` according to user guide
* Remove ml-cache from run_slurm.sh (see https://github.com/groq/mlagility/pull/79#discussion_r1101536383 for details)

### PR DESCRIPTION
**Pull request by danielholanda**
This MR deduplicates the slurm infrastructure and adds support for `--use-slurm` on mlagility.
Closes #89

### CODE REVIEW

File: src/mlagility/cli/run_slurm.sh [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 export XDG_CACHE_HOME="/net/ml-cache/huggingface"
 export TORCH_HOME="/net/ml-cache/torch-hub"
-
-# shellcheck source=/dev/null
-source /net/home/"$USER"/miniconda3/etc/profile.d/conda.sh
-conda activate groqit_slurm
-export USING_SLURM="TRUE"
-# shellcheck disable=SC2046
-# shellcheck disable=SC2116
+export TORCH_HUB="/net/ml-cache/torch-hub"
```
Comment (jeremyfowers) [inline src/mlagility/cli/run_slurm.sh:15]: These kind of exports are pretty spicy for an open source project. Most users wont have this directory. Maybe these exports just need to go into the .bashrc files of Groq developers?

Comment (danielholanda) [reply src/mlagility/cli/run_slurm.sh]: Agreed. I'm only deduplicating the infrastructure here. Added that as a todo of https://github.com/groq/mlagility/issues/89

File: docs/install.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# MlAgility Installation Guide
```
Comment (jeremyfowers) [inline docs/install.md:1]: Link on //readme.md and the benchit user guide

File: docs/install.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Install the `mlagility` package into your environment simply run
+
+```
+pip install -e .
+```
+
+inside the mlagility repository.
+
+## Installing Slurm support
```
Comment (jeremyfowers) [inline docs/install.md:15]: Say what slurm is and why someone would want to install slurm support

Comment (jeremyfowers) [reply docs/install.md:15]: And link to the `--use-slurm` documentation as well

**Comment by jeremyfowers** [comment]
Code looks good to me, but there seems to be a bug:

```
(mla) jfowers@jfowers:~/mlagility$ benchit benchmark --all -s models/diffusers/ --use-slurm --build-only --cache-dir /net/home/jfowers/Groq/mlagility_cache

Submitting job unet_2d_condition to Slurm
Submitted batch job 2134033
Submitting job vae_decoder to Slurm
Submitted batch job 2134034
Submitting job clip_text_encoder to Slurm
Submitted batch job 2134035
Submitting job safety_clipvision to Slurm
Submitted batch job 2134036
Waiting: 4 jobs left in queue: ['unet_2d_', 'vae_deco', 'clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']
Waiting: 2 jobs left in queue: ['clip_tex', 'safety_c']

Woohoo! The 'benchmark' command is complete. Use the 'report' command to get a .csv file that summarizes results across all builds in the cache.
```

followed by

```
(mla) jfowers@jfowers:~/mlagility$ benchit cache list --cache-dir /net/home/jfowers/mlagility_cache

Info: Builds available in cache /net/home/jfowers/mlagility_cache:
Traceback (most recent call last):
  File "/net/home/jfowers/miniconda3/envs/mla/bin/benchit", line 33, in <module>
    sys.exit(load_entry_point('mlagility', 'console_scripts', 'benchit')())
  File "/home/jfowers/mlagility/src/mlagility/cli/cli.py", line 406, in main
    args.func(args)
  File "/home/jfowers/mlagility/src/mlagility/common/filesystem.py", line 116, in print_available_builds
    builds = get_available_builds(args.cache_dir)
  File "/home/jfowers/mlagility/src/mlagility/common/filesystem.py", line 106, in get_available_builds
    for build in os.listdir(os.path.abspath(cache_dir))
FileNotFoundError: [Errno 2] No such file or directory: '/net/home/jfowers/mlagility_cache'
```

It needs to mkdir the cache dir

**Comment by jeremyfowers** [comment]
following up on https://github.com/groq/mlagility/pull/79#issuecomment-1426196479, if I manually mkdir the cache directory, it is empty at the end of running `benchit benchmark --all -s models/diffusers/ --use-slurm --build-only --cache-dir /net/home/jfowers/Groq/mlagility_cache`

I also dont seem to have a slurm log anywhere.

File: docs/install.md (hunk 2 above)
Comment (danielholanda) [reply docs/install.md:1]: Added!

File: docs/install.md (hunk 3 above)
Comment (danielholanda) [reply docs/install.md:15]: Updated!

**Comment by danielholanda** [comment]
@jeremyfowers All looking good now. Please try running `benchit benchmark --all -s models/diffusers/ --use-slurm --build-only --cache-dir /net/home/jfowers/Groq/mlagility_cache` again.

**Pull request merged**